### PR TITLE
Fix expiry date related bugs

### LINKED ISSUE #167: [PE-D] Inconsistent error messages when adding invalid expiry dates
I am able to create food items from millenia ago. There could be more documentation on the valid expiry dates.

However I am unable to create food from year 0000

How to recreate: simply try adding an item with expiry date `01-01-0000`.

If this is intended behaviour, the error message is also inconsistent with the other error messages.

or

-------------
Labels: `severity.VeryLow` `type.DocumentationBug`
original: JoeyChenSmart/ped#2

### PR DESCRIPTION
**Pull request by songyiang**
Error messages for invalid expiry dates have been standardised.

The year of expiry date is now constrained between 2020 and 2120, both
inclusive.

More test cases for expiry date have been added.

### CODE REVIEW

File: src/main/java/seedu/simplykitchen/model/food/ExpiryDate.java [hunk 1]
```diff
@@ -16,9 +16,8 @@
 public class ExpiryDate {
 
     public static final String MESSAGE_CONSTRAINTS =
-            "The expiry date should be of the format DD-MM-YYYY or DD/MM/YYYY. ";
-    public static final String MESSAGE_SHORTENED_YEAR = "The year should be 4 digits long. ";
-    public static final String MESSAGE_INVALID_DATE = "The expiry date does not exist. ";
+            "The expiry date should be of the format DD-MM-YYYY or DD/MM/YYYY. "
+                    + "The year should be between 2020 and 2120, both inclusive.";
```
Comment (bchenghi) [inline src/main/java/seedu/simplykitchen/model/food/ExpiryDate.java:20]: Should we include an error message for invalid date? For example 29-2-2021 is the correct format and is between 2020 and 2120, but is not valid.

**Review by bchenghi — COMMENTED** [review]
The changes look good! I added one comment.

File: src/main/java/seedu/simplykitchen/model/food/ExpiryDate.java (hunk 1 above)
Comment (songyiang) [reply src/main/java/seedu/simplykitchen/model/food/ExpiryDate.java:20]: I edited the message to include the valid date part.

The two date formatters used check for different types of formatting errors and also invalid date as well. Hence, the formatters can't differentiate between an invalid date or a date with incorrect formatting.

Comment (bchenghi) [reply src/main/java/seedu/simplykitchen/model/food/ExpiryDate.java:20]: Oh I see. Thanks!

**Review by bchenghi — APPROVED** [review]
LGTM

**Pull request merged**